tests/run/test_genericclass.py

```diff
@@ -0,0 +1,288 @@
+### COPIED FROM CPython 3.7
```

Comment:
On the first line, add:

```
#mode: run
#tag: pure3.7
```

to tell the testrunner that a) the test should be run, and b) it should also run it in plain Python>=3.7 just to show that it matches the plain Python behaviour